docs/prover/approx/hashing.md

```diff
@@ -0,0 +1,182 @@
+Modeling of Hashing in CVT
+===========
+
+
+## Introduction
+
+The Keccak hash function is used heavily by Solidity smart contracts. 
+Most prominently, all unbounded data structures in storage (arrays, mappings) receive their storage addresses as values of the Keccak function.
+
+The Certora Prover does not operate with an actual implementation of the Keccak hash function, since this would make most verification intractable and provide no benefits in all practical cases we are aware of so far.
+Instead, the Keccak hash function is modeled as an arbitrary function that is _injective with large gaps_. 
+
+The hash function `hash` being injective with large gaps means that on distinct inputs `x` and `y`
+  - the hashes `hash(x)` and `hash(y)` are also distinct, and
+  - the gap between `hash(x)` and `hash(y)` is large enough that every additive term `hash(x) + o` that occurs in the program is also distinct from `hash(y)`.
+
+## Background: The Solidity Storage Model
```

Comment:
This section seems pretty heavy, and not super-relevant to users.  Maybe omit, or more clearly indicate how this might affect the user.

Reply:
hm, I could only reference it here, and push it to the end (?)
I can also extend the example with a rule that relies on the non-interference ..

Reply:
(done that)